core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -166,6 +170,48 @@ public B defaultMaxResponseLength(long defaultMaxResponseLength) {
         return option(ClientOption.DEFAULT_MAX_RESPONSE_LENGTH, defaultMaxResponseLength);
     }
 
+    /**
+     * Sets the {@link ContentPreviewerFactory} for a request.
+     */
+    public B requestContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return option(ClientOption.REQ_CONTENT_PREVIEWER_FACTORY,
+                      requireNonNull(factory, "factory"));
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} for a response.
+     */
+    public B responseContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return option(ClientOption.RES_CONTENT_PREVIEWER_FACTORY,
+                      requireNonNull(factory, "factory"));
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} for a request and a response.
+     */
+    public B contentPreviewerFactory(ContentPreviewerFactory factory) {
+        requireNonNull(factory, "factory");
+        requestContentPreviewerFactory(factory);
+        responseContentPreviewerFactory(factory);
+        return self();
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} creating a {@link ContentPreviewer} which produces the preview
+     * with the maxmium {@code length} limit for a request and a response.
+     */
+    public B contentPreview(int length, Charset defaultCharset) {
+        return contentPreviewerFactory(ContentPreviewerFactory.ofText(length, defaultCharset));
+    }
+
+    /**
+     * Sets the {@link ContentPreviewerFactory} creating a {@link ContentPreviewer} which produces the preview
+     * with the maxmium {@code length} limit for a request and a response.
```

Comment:
How about adding an extra sentence that explains what content types are supported?